src/core/chunked_stream.js

```diff
@@ -290,283 +276,274 @@ var ChunkedStreamManager = (function ChunkedStreamManagerClosure() {
+    }
+    return this._requestChunks(chunks);
+  }
 
-      for (var i = 0; i < ranges.length; i++) {
-        var beginChunk = this.getBeginChunk(ranges[i].begin);
-        var endChunk = this.getEndChunk(ranges[i].end);
-        for (var chunk = beginChunk; chunk < endChunk; ++chunk) {
-          if (!chunksToRequest.includes(chunk)) {
-            chunksToRequest.push(chunk);
-          }
+  requestRanges(ranges = []) {
+    const chunksToRequest = [];
+    for (let i = 0; i < ranges.length; i++) {
+      const beginChunk = this.getBeginChunk(ranges[i].begin);
+      const endChunk = this.getEndChunk(ranges[i].end);
+      for (let chunk = beginChunk; chunk < endChunk; ++chunk) {
+        if (!chunksToRequest.includes(chunk)) {
+          chunksToRequest.push(chunk);
         }
       }
+    }
 
-      chunksToRequest.sort(function(a, b) {
-        return a - b;
-      });
-      return this._requestChunks(chunksToRequest);
-    },
-
-    // Groups a sorted array of chunks into as few contiguous larger
-    // chunks as possible
-    groupChunks: function ChunkedStreamManager_groupChunks(chunks) {
-      var groupedChunks = [];
-      var beginChunk = -1;
-      var prevChunk = -1;
-      for (var i = 0; i < chunks.length; ++i) {
-        var chunk = chunks[i];
-
-        if (beginChunk < 0) {
-          beginChunk = chunk;
-        }
+    chunksToRequest.sort(function(a, b) {
+      return a - b;
+    });
+    return this._requestChunks(chunksToRequest);
+  }
 
-        if (prevChunk >= 0 && prevChunk + 1 !== chunk) {
-          groupedChunks.push({ beginChunk,
-                               endChunk: prevChunk + 1, });
-          beginChunk = chunk;
-        }
-        if (i + 1 === chunks.length) {
-          groupedChunks.push({ beginChunk,
-                               endChunk: chunk + 1, });
-        }
+  /**
+   * Groups a sorted array of chunks into as few contiguous larger
+   * chunks as possible.
+   */
+  groupChunks(chunks) {
+    const groupedChunks = [];
+    let beginChunk = -1;
+    let prevChunk = -1;
 
-        prevChunk = chunk;
-      }
-      return groupedChunks;
-    },
-
-    onProgress: function ChunkedStreamManager_onProgress(args) {
-      var bytesLoaded = (this.stream.numChunksLoaded * this.chunkSize +
-                         args.loaded);
-      this.msgHandler.send('DocProgress', {
-        loaded: bytesLoaded,
-        total: this.length,
-      });
-    },
-
-    onReceiveData: function ChunkedStreamManager_onReceiveData(args) {
-      var chunk = args.chunk;
-      var isProgressive = args.begin === undefined;
-      var begin = isProgressive ? this.progressiveDataLength : args.begin;
-      var end = begin + chunk.byteLength;
-
-      var beginChunk = Math.floor(begin / this.chunkSize);
-      var endChunk = end < this.length ? Math.floor(end / this.chunkSize) :
-                                         Math.ceil(end / this.chunkSize);
+    for (let i = 0; i < chunks.length; ++i) {
```

Comment:
Could potentially use `for (const chunk of chunks) {` here instead.

Reply:
Unfortunately not in this case, because the variable `i` is used a few lines below.